src/app/core/services/phase.service.ts

```diff
@@ -160,6 +160,16 @@ export class PhaseService {
    * and synchronized with the remote server.
    */
   sessionSetup(): Observable<any> {
+    // Permission Caching Mechanism to prevent repeating permission request.
+    let isSessionFixPermissionGranted = false;
+    const cacheSessionFixPermission = () => {
+      return pipe(
+        tap((sessionFixPermission: boolean | null) => {
+          isSessionFixPermissionGranted = sessionFixPermission ? sessionFixPermission : false;
+        })
+      );
+    };
```

Comment:
@anubh-v 

I thought perhaps by introducing a local function which uses pipe, the main sessionSetup part might be abit neater following the #549 .

Would this be okay?

Reply:
Yes, this works cleanly :+1: